File: Python/Important-Programs/22-automorphic-number-or-not.py
num = int(input("Enter a number: "))
square = pow(num, 2)
mod = pow(10, len(str(num)))

if (square%mod == num):
    print(num, "is Automorphic Number.")
else:
    print(num, "is Not Automorphic Number.")


# Using enswith() method
# a = str(num)
# b = str(square)

# if b.endswith(a):
#     print("Automorphic number")
# else:
#     print("Not Automorphic number")

File: Java/Important-Programs/22-automorphic-number-or-not.java
import java.util.Scanner;

class Main {
    public static void main(String[] args) {
        Scanner getInput = new Scanner(System.in);

        System.out.println("Enter a number:");
        int num = getInput.nextInt();
        
        int square = num*num;
        int length = String.valueOf(num).length();
        int mod =  (int) Math.pow(10, length);

        if (square%mod == num){
            System.out.println(num + " is Automorphic number.");
        } else {
            System.out.println(num + " is Not Automorphic number.");
        }

        getInput.close();
    }
}